lib/pebblebed/http.rb

```diff
@@ -191,44 +218,56 @@ def self.handle_http_errors(response)
       response
     end
 
-    def self.do_easy(cache: true, &block)
-      with_easy(cache: cache) do |easy|
-        yield easy
-        response = Response.new(easy.url, easy.header_str, easy.body_str)
-        return handle_http_errors(response)
-      end
+    def self.do_request(url, &block)
+      Timeout::timeout(request_timeout, Excon::Errors::Timeout) {
```

Comment:
Never use `timeout` ([see this article](http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/)). Excon implements timeouts on its own, so I don't think this should be necessary?

Reply:
Removed the timeout!